pkg/grpc/api.go

```diff
@@ -787,6 +795,119 @@ func (a *api) getStateStore(name string) (state.Store, error) {
 	return a.stateStores[name], nil
 }
 
+func (a *api) getComponent(componentKind string, componentName string) (component interface{}) {
+	switch componentKind {
+	case "state":
```

Comment:
This API is currently supporting only state, pubsub, secretstore and bindings. 
For now, for any other component type, it will give ERR_PING_NOT_IMPLEMENTED. Later, we can definitely include other components as well.